### PR TITLE
Use Kustomize to generate test templates

### PR DESCRIPTION
**Pull request by sedefsavas**
**What type of PR is this?**
This PR adds a script to build test manifests using Kustomize (using a similar script with [CAPZ](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/hack/gen-flavors.sh))

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2134

**Special notes for your reviewer**:

**Checklist**:

- [X] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [X] adds or updates e2e tests

**Release note**:

```release-note
Add e2e test for using external cloud controller manager 
```

### CODE REVIEW

**Comment by sedefsavas** [comment]
`Upgrade to master` e2e test fails due to trying to upgrade 2 versions ahead:
`Kubernetes version from v1.20.4 to v1.22.0-alpha.0.349+46563b0abebbb0`
This is fixed in this PR: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2253

External cloud provider e2e test is passing, so this PR is ready to review.

cc @randomvariable

**Comment by randomvariable** [comment]
It's more important we solve #2131 first otherwise we are going to be waiting forever.

Also, do we always need a HA cluster test?

**Comment by randomvariable** [comment]
needs rebasing.

do you want to give me a preferred order for PRs to merge in?

**Comment by sedefsavas** [comment]
Let's wait until e2e parallelization pr is merged: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2382
/hold

File: Makefile [hunk 1]
```diff
@@ -146,6 +146,10 @@ $(ARTIFACTS):
 test: ## Run tests
 	source ./scripts/fetch_ext_bins.sh; fetch_tools; setup_envs; go test -v ./...
 
+.PHONY: generate-test-flavors ## Generate test template flavors
+generate-test-flavors: $(KUSTOMIZE)
```
Comment (randomvariable) [inline Makefile:150]:
```suggestion
.PHONY: generate-test-flavors
generate-test-flavors: $(KUSTOMIZE)  ## Generate test template flavors
```

Typo for the help

**Review by randomvariable — COMMENTED** [review]
This is otherwise fine, but it would be super useful to add a README.md to test/e2e/data/infrastructure-aws explaining how  this works. It's not immediately obvious that you use should use kustomize for adding a template given that these are checked into the repo.

Consider deleting the generated files and adding them to a test/e2e/data/infrastructure-aws/.gitignore too.

**Comment by sedefsavas** [comment]
Adding explanation to the book.

> Consider deleting the generated files and adding them to a test/e2e/data/infrastructure-aws/.gitignore too

I think it is useful to keep the files, as in our templates, we don't keep all examples we have in the tests. CAPZ is also keeping them.

**Pull request merged**